src/svg_to_png/lib/ThreatModel.py

```diff
@@ -86,6 +89,16 @@ def set_groups(nodes, boundaries):
         for boundary in boundaries:
             set_appropriate_groups(node, boundary)
 
+def set_curve_nodes(nodes, curves):
+    for curve in curves:
+        for node in nodes:
+            if is_point_in(curve.sourceX, curve.sourceY, node):
+                curve.sourceNode = node
+            elif is_point_in(curve.targetX, curve.targetY, node):
```

Comment:
a node can potentially point to itself too

Reply:
Good point! Updated.